BaikalDataGenerator/Source/render.cpp

```diff
@@ -99,52 +105,51 @@ Render::Render(const std::string &file_name,
     m_renderer = m_factory->CreateRenderer(Baikal::ClwRenderFactory::RendererType::kUnidirectionalPathTracer);
     m_controller = m_factory->CreateSceneController();
 
-    for (const auto output_info: outputs_collection)
+    for (auto& output_info: outputs_collection)
     {
         m_outputs.push_back(m_factory->CreateOutput(output_width, output_height));
         m_renderer->SetOutput(output_info.type, m_outputs.back().get());
+        output_info.width = output_width;
+        output_info.height = output_height;
     }
 
+    std::replace(file_name.begin(), file_name.end(), '\\', '/');
+    size_t last_index = file_name.find_last_of('/');
+    std::string path = file_name.substr(0, last_index);
+
     m_scene = (file_name.empty() || path.empty()) ? Baikal::SceneIo::LoadScene("sphere+plane.test", "") :
                                                     Baikal::SceneIo::LoadScene(file_name, path);
 }
 
-void Render::LoadMaterialXml(const std::string &path, const std::string &file_name)
+void Render::LoadMaterialXml(const std::string &file_name)
 {
-    std::stringstream ss;
-    ss << path << "/" << file_name;
-
     // Check it we have material remapping
-    std::ifstream in_materials(ss.str());
-    std::ifstream in_mapping(ss.str());
+    std::ifstream in_materials(file_name);
+    std::ifstream in_mapping(file_name);
 
     if (in_materials && in_mapping)
     {
         in_materials.close();
         in_mapping.close();
 
         auto material_io = Baikal::MaterialIo::CreateMaterialIoXML();
-        auto mats = material_io->LoadMaterials(ss.str());
-        auto mapping = material_io->LoadMaterialMapping(ss.str());
+        auto mats = material_io->LoadMaterials(file_name);
+        auto mapping = material_io->LoadMaterialMapping(file_name);
 
         material_io->ReplaceSceneMaterials(*m_scene, *mats, mapping);
```

Comment:
We should check that materials and mapping are loaded before using them

Reply:
removed